File: integrationtest/src/main/java/com/sequenceiq/it/cloudbreak/config/ITProps.java
package com.sequenceiq.it.cloudbreak.config;

import java.util.List;
import java.util.Map;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "integrationtest")
public class ITProps {
    private Map<String, String> credentialNames;
    private Map<String, List<String>> testSuites;
    private List<String> testTypes;
    private List<String> suiteFiles;

    public void setCredentialNames(Map<String, String> credentialNames) {
        this.credentialNames = credentialNames;
    }

    public Map<String, String> getCredentialNames() {
        return credentialNames;
    }

    public String getCredentialName(String cloudProvider) {
        return credentialNames.get(cloudProvider);
    }

    public void setTestSuites(Map<String, List<String>> testSuites) {
        this.testSuites = testSuites;
    }

    public Map<String, List<String>> getTestSuites() {
        return testSuites;
    }

    public List<String> getTestSuites(String suitesKey) {
        return testSuites.get(suitesKey);
    }

    public List<String> getTestTypes() {
        return testTypes;
    }

    public void setTestTypes(List<String> testTypes) {
        this.testTypes = testTypes;
    }

    public List<String> getSuiteFiles() {
        return suiteFiles;
    }

    public void setSuiteFiles(List<String> suiteFiles) {
        this.suiteFiles = suiteFiles;
    }
}

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/config/ITProps.kt
package com.sequenceiq.it.cloudbreak.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "integrationtest")
class ITProps {
    var credentialNames: Map<String, String>? = null
    var defaultNetworks: Map<String, String>? = null
    var testSuites: Map<String, List<String>>? = null
    var testTypes: List<String>? = null
    var suiteFiles: List<String>? = null
    var defaultSecurityGroup: String? = null

    fun getCredentialName(cloudProvider: String): String {
        return credentialNames!![cloudProvider]
    }

    fun getDefaultNetwork(cloudProvider: String): String {
        return defaultNetworks!![cloudProvider]
    }

    fun getTestSuites(suitesKey: String): List<String> {
        return testSuites!![suitesKey]
    }
}
